### DISCUSSION TITLE
Set audio and scaling via environment variables?

### DISCUSSION THREAD

**Discussion by MrCussMustard** [jlesage/docker-firefox#254] [Q&A] [answered]
The vnc always launches with audio muted and often enough it loses track of the scaling settings -- is there a way to set scaling and sound volume via environment variable? Soundrecorder service is activated when I set the volume to antything other than default, so must I evable this service?

Essentially, I don't want to have to touch the settings menu when I start browsing. Any advice?

**Comment by jlesage** [comment]
> The vnc always launches with audio muted

This is expected: browsers don't allow a web page to automatically play sound without user action.

> often enough it loses track of the scaling settings 

Scaling settings are saved by the browser. So if you use the same browser and don't clear the cache/cookies, the settings should be kept. Could you provide more details about the conditions required to reproduce this problem?

**Reply by MrCussMustard** [reply]
Losing the scaling then likely has to do with my constant clearing of cache and cookies while I set up a nativefied Organizr and is therefore a non-issue. Thank you.

Do you have advice on automating the process to unmute? Am I missing your point, or could i not simply write a script to execute the same command that is issued when I click the muted speaker icon and it sets the volume to 50? Essentially tricking the vnc into thinking I clicked it?

I guess what I am asking is what command the volume bar executes, and if I should assume it is impossible to script that event.

**Reply by jlesage** [reply, marked as answer]
To my knowledge, the problem is that the audio API must first be called from a callback triggered by a user input (e.g. a click). In other words, if you call, when loading the page, the same function that is called when clicking the unmute button, it won't work. This is a protection added by browsers to prevent a page from playing audio automatically, which would be problematic for users!
